docs/rules/no-unresolved.md

```diff
@@ -53,6 +53,16 @@ define(['./foo'], function (foo) { /*...*/ }) // reported if './foo' is not foun
 require(['./foo'], function (foo) { /*...*/ }) // reported if './foo' is not found
 ```
 
+If `{requireResolve: true}` is provided, single-argument `require.resolve` calls will be resolved:
+
+```js
+/*eslint import/no-unresolved: [2, { requireResolve: true }]*/
+const { default: x } = require.resolve('./foo') // reported if './foo' is not found
+
+require.resolve(0) // ignored
```

Comment:
why is this ignored? i'd expect this to be coerced to a string and then validated the same.

Reply:
Attempting to run `require.resolve(0)` gives me this stacktrace:

```bash
TypeError [ERR_INVALID_ARG_TYPE]: The "request" argument must be of type string. Received type number
    at Function.resolve (internal/modules/cjs/helpers.js:28:13)
```

Reply:
it seems useful to me to have the linter warn on it

Reply:
I agree -- but we should follow the same rules we have for `commonjs` `require` right?

Reply:
Not necessarily - `require` is used in multiple ways, and so has edge cases - `require.resolve`, however, does not have those edge cases.

Reply:
I'm not sure what you mean. I've never run into any cases where the arguments of `require` (commonjs) and `require.resolve` work differently.

From [the node documentation](https://nodejs.org/api/modules.html#modules_all_together) it says that `require.resolve` "gets the exact filename that will be loaded when `require()`".